motr/m0crate/crate_index.c

```diff
@@ -648,14 +780,16 @@ static int fill_kv_put(struct cr_idx_w *w,
 	M0_ASSERT(p->v->ov_vec.v_nr == nr);
 
 	for (i = 0; i < nr; i++) {
-		p->k->ov_vec.v_count[i] = sizeof(*k);
-		p->k->ov_buf[i] = m0_alloc_aligned(sizeof(*k), PAGE_SHIFT);
-		memcpy(p->k->ov_buf[i], &k[i], sizeof(*k));
+		p->k->ov_vec.v_count[i] = w->wit->key_size;
+		p->k->ov_buf[i] = m0_alloc_aligned(w->wit->key_size, PAGE_SHIFT);
+		memcpy(p->k->ov_buf[i], (void*)kpart_one, kpart_one_size);
+		memcpy(p->k->ov_buf[i] + kpart_one_size, &k[i], sizeof(*k));
```

Comment:
Can we re-factor the code into a function call?
This piece of code is duplicated in multiple places.

Reply:
I would like to recommend, we can keep as earlier, to keep it more readable.